src/gc/impl/conservative/gc.d

```diff
@@ -332,6 +323,7 @@ class ConservativeGC : GC
             cstdlib.free(gcx);
             gcx = null;
         }
+        cstdlib.free(cast(void*) this);
```

Comment:
Freeing `this` in a public method is a bit dangerous because an invariant might crash. One possible solution might be to pass the finalize function to the registry, too.

Reply:
Yes, not so nice.

In fact just using a static buffer instead of the C heap might work as well.

Think the best alternative is to convert Dtor to a real destructor which doesn't have the invariant problem.